qa/tasks/cephfs/test_mirroring.py

```diff
@@ -625,3 +647,46 @@ def test_cephfs_mirror_failed_sync_with_correction(self):
         self.check_peer_status(self.primary_fs_name, self.primary_fs_id,
                                "client.mirror_remote@ceph", '/d0', 'snap0', 1)
         self.disable_mirroring(self.primary_fs_name, self.primary_fs_id)
+
+    def test_cephfs_mirror_service_daemon_status(self):
```

Comment:
````
2021-03-04T11:44:34.995 INFO:tasks.cephfs_test_runner:======================================================================
2021-03-04T11:44:34.996 INFO:tasks.cephfs_test_runner:FAIL: test_cephfs_mirror_service_daemon_status (tasks.cephfs.test_mirroring.TestMirroring)
2021-03-04T11:44:34.996 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2021-03-04T11:44:34.996 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2021-03-04T11:44:34.997 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_d5875db5f1fdb7ac11e6c9da991269513ffdac1a/qa/tasks/cephfs/test_mirroring.py", line 678, in test_cephfs_mirror_service_daemon_status
2021-03-04T11:44:34.997 INFO:tasks.cephfs_test_runner:    self.assertEquals(peer_stats['failure_count'], 1)
2021-03-04T11:44:34.997 INFO:tasks.cephfs_test_runner:AssertionError: 0 != 1
````

From: /ceph/teuthology-archive/pdonnell-2021-03-04_03:51:01-fs-wip-pdonnell-testing-20210303.195715-distro-basic-smithi/5932070/teuthology.log

Reply:
It's a timing thing. The mirror daemon marks a directory as "failed" after it has tried synchronization (w/ failure) seeing "N" consecutive failures.

Reply:
From logs:

```
2021-03-04T11:43:04.557+0000 7f3c0c298700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
2021-03-04T11:43:14.558+0000 7f3c0c298700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
2021-03-04T11:43:24.560+0000 7f3c0ba97700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
2021-03-04T11:43:34.562+0000 7f3c0ba97700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
2021-03-04T11:43:44.565+0000 7f3c0b296700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
2021-03-04T11:43:54.566+0000 7f3c0c298700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
2021-03-04T11:44:04.568+0000 7f3c0c298700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
2021-03-04T11:44:14.569+0000 7f3c0b296700 -1 cephfs::mirror::PeerReplayer(efca7e05-a062-4eb6-99e5-d555726d6052) sync_snaps: failed to sync snapshots for dir_path=/d0
```

For peer uuid: efca7e05-a062-4eb6-99e5-d555726d6052, the mirror daemon has not yet hit the consecutive failure count (default: 10).

Reply:
Add a `safe_while` to test?

Reply:
I avoided that since it repeatedly runs a command (although the delay is configurable). We already kind of know the expected wait time here before a directory is marked as failed.

I can convert all the wait calls to use safe_while in another PR since it requires an audit of the tests.